tests/integration_tests/test_gym_envs.py

```diff
@@ -0,0 +1,130 @@
+from itertools import product
+from multiprocessing import Process
+import pickle
+import unittest
+
+import gym
+from nose2.tools import params
+
+from garage.tf.envs.base import TfEnv
+from garage.theano.envs.base import TheanoEnv
+
+# White list of environments that cannot be rendered
+not_rendering_envs = [
+    "Blackjack-v0", "NChain-v0", "Roulette-v0", "GuessingGame-v0",
+    "HotterColder-v0", "Blackjack-v0", "NChain-v0", "Roulette-v0",
+    "GuessingGame-v0", "HotterColder-v0"
+]
+
+# White list of environments that cannot be closed
+not_closing_envs = [
+    "Reacher-v2", "Pusher-v2", "Thrower-v2", "Striker-v2",
+    "InvertedPendulum-v2", "InvertedDoublePendulum-v2", "HalfCheetah-v2",
+    "Hopper-v2", "Swimmer-v2", "Walker2d-v2", "Ant-v2", "Humanoid-v2",
+    "HumanoidStandup-v2", "FetchSlide-v1", "FetchPickAndPlace-v1",
+    "FetchReach-v1", "FetchPush-v1", "HandReach-v0",
+    "HandManipulateBlockRotateZ-v0", "HandManipulateBlockRotateParallel-v0",
+    "HandManipulateBlockRotateXYZ-v0", "HandManipulateBlockFull-v0",
+    "HandManipulateBlock-v0", "HandManipulateEggRotate-v0",
+    "HandManipulateEggFull-v0", "HandManipulateEgg-v0",
+    "HandManipulatePenRotate-v0", "HandManipulatePenFull-v0",
+    "HandManipulatePen-v0", "FetchSlideDense-v1", "FetchPickAndPlaceDense-v1",
+    "FetchReachDense-v1", "FetchPushDense-v1", "HandReachDense-v0",
+    "HandManipulateBlockRotateZDense-v0",
+    "HandManipulateBlockRotateParallelDense-v0",
+    "HandManipulateBlockRotateXYZDense-v0", "HandManipulateBlockFullDense-v0",
+    "HandManipulateBlockDense-v0", "HandManipulateEggRotateDense-v0",
+    "HandManipulateEggFullDense-v0", "HandManipulateEggDense-v0",
+    "HandManipulatePenRotateDense-v0", "HandManipulatePenFullDense-v0",
+    "HandManipulatePenDense-v0"
+]
+
+
+class TestGymEnvs(unittest.TestCase):
+
+    garage_envs = [TfEnv, TheanoEnv]
+
+    def _routine_test_pickling_wrapper(self, garage_env, spec):
```

Comment:
If you need a fixture, please put it in `tests.fixtures`. See the env test suite I made in `env_pickle_tests`